0.21 - Crash on open

Application crashes on update of screen after open. 

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x7ff628e4c552]

goroutine 136 [running]:
github.com/ErikKalkoken/evebuddy/internal/app.EveLocation.DisplayRichText({0xf2d6025309, 0x0, 0x0, {0x0, 0x0}, 0x0, {0x28caa1ac, 0xedf6631d7, 0xc00041abd0}})
	github.com/ErikKalkoken/evebuddy/internal/app/evelocation.go:52 +0xb2
github.com/ErikKalkoken/evebuddy/internal/app/ui/characteroverview.NewAssets.func3(0xc005425ab0?, 0x7ff62837b4f3?)
	github.com/ErikKalkoken/evebuddy/internal/app/ui/characteroverview/assets.go:102 +0xdf
github.com/ErikKalkoken/evebuddy/internal/widget.MakeDataTableForDesktop2[...].func3({0x7ff62a77a020?, 0xc001bfbd40?})
	github.com/ErikKalkoken/evebuddy/internal/widget/tableHelper.go:164 +0x58
fyne.io/fyne/v2/widget.(*tableCellsRenderer).refreshForID(0xc0003a8000, {0x28a02de5?, 0xc0007d43c0?})
	fyne.io/fyne/v2@v2.5.5/widget/table.go:1403 +0xbba
fyne.io/fyne/v2/widget.(*tableCellsRenderer).Refresh(0x7ff62a77e978?)
	fyne.io/fyne/v2@v2.5.5/widget/table.go:1263 +0x25
fyne.io/fyne/v2/widget.(*BaseWidget).Refresh(0xc0007d43c0)
	fyne.io/fyne/v2@v2.5.5/widget/widget.go:137 +0x9d
fyne.io/fyne/v2/widget.(*tableRenderer).Refresh(0xc0003f48a0)
	fyne.io/fyne/v2@v2.5.5/widget/table.go:1127 +0x185
fyne.io/fyne/v2/widget.(*BaseWidget).Refresh(0xc00132e180)
	fyne.io/fyne/v2@v2.5.5/widget/widget.go:137 +0x9d
github.com/ErikKalkoken/evebuddy/internal/app/ui/characteroverview.(*Assets).Update(0xc001bc0d20)
	github.com/ErikKalkoken/evebuddy/internal/app/ui/characteroverview/assets.go:273 +0x16a
github.com/ErikKalkoken/evebuddy/internal/app/ui.runFunctionsWithProgressModal.func1.1()
	github.com/ErikKalkoken/evebuddy/internal/app/ui/ui.go:468 +0xcb
created by github.com/ErikKalkoken/evebuddy/internal/app/ui.runFunctionsWithProgressModal.func1 in goroutine 1028
	github.com/ErikKalkoken/evebuddy/internal/app/ui/ui.go:465 +0x1a6

--- a/FyneApp.toml
+++ b/FyneApp.toml
@@ -4,7 +4,7 @@ Website = "https://github.com/ErikKalkoken/evebuddy"
   Icon = "icon.png"
   Name = "EVE Buddy"
   ID = "io.github.erikkalkoken.evebuddy"
-  Version = "0.21.0"
+  Version = "0.21.1"
   Build = 0
 
 [Release]

--- a/internal/app/evelocation.go
+++ b/internal/app/evelocation.go
@@ -48,12 +48,15 @@ func (el EveLocation) DisplayRichText() []widget.RichTextSegment {
 	} else {
 		n = el.alternativeName()
 	}
+	if el.SolarSystem == nil {
+		return iwidget.NewRichTextSegmentFromText(n)
+	}
 	return slices.Concat(
 		el.SolarSystem.SecurityStatusRichText(),
 		iwidget.NewRichTextSegmentFromText(fmt.Sprintf("  %s", n)))
 }
 
-// DisplayName2 returns a user friendly name not including the location name.
+// DisplayName2 returns a user friendly name not including the sytem name.
 func (el EveLocation) DisplayName2() string {
 	if el.Name != "" {
 		if el.Variant() != EveLocationStructure {

--- a/internal/app/evelocation_test.go
+++ b/internal/app/evelocation_test.go
@@ -4,6 +4,8 @@ import (
 	"fmt"
 	"testing"
 
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
 	"github.com/ErikKalkoken/evebuddy/internal/app"
 	"github.com/stretchr/testify/assert"
 )
@@ -26,7 +28,7 @@ func TestLocationVariantFromID(t *testing.T) {
 	}
 }
 
-func TestEveLocation(t *testing.T) {
+func TestEveLocationDisplayName2(t *testing.T) {
 	cases := []struct {
 		in  string
 		out string
@@ -44,4 +46,35 @@ func TestEveLocation(t *testing.T) {
 			assert.Equal(t, tc.out, x.DisplayName2())
 		})
 	}
+}
+
+func TestEveLocationDisplayRichText(t *testing.T) {
+	t.Run("can return as rich text", func(t *testing.T) {
+		ss := &app.EveSolarSystem{SecurityStatus: 0.5}
+		l := &app.EveLocation{Name: "location_name", SolarSystem: ss}
+		got := l.DisplayRichText()
+		want := []widget.RichTextSegment{
+			&widget.TextSegment{
+				Text: "0.5",
+				Style: widget.RichTextStyle{
+					ColorName: theme.ColorNameSuccess,
+					Inline:    true,
+				},
+			},
+			&widget.TextSegment{
+				Text: "  location_name",
+			},
+		}
+		assert.Equal(t, want, got)
+	})
+	t.Run("can handle missing solar system", func(t *testing.T) {
+		l := &app.EveLocation{Name: "location_name"}
+		got := l.DisplayRichText()
+		want := []widget.RichTextSegment{
+			&widget.TextSegment{
+				Text: "location_name",
+			},
+		}
+		assert.Equal(t, want, got)
+	})
 }

